gloo/docs/enterprise/authentication/custom_auth/_index.md

```diff
@@ -0,0 +1,172 @@
+    socketserver.TCPServer(('', port), Server).serve_forever()
+
+if __name__ == "__main__":
+    serve_forever(8000)
+```
+
+As you can see, this service will allow requests to `/api/pets/1` and will deny everything else.
+
+{{% notice tip %}}
+You can easily change the sample auth server. When using minikube, download the [Dockerfile](Dockerfile) and the [server code](server.py) and just run:
+
+```shell
+eval $(minikube docker-env)
+docker build -t quay.io/solo-io/sample-auth .
+kubectl delete pod -n gloo-system -l app=sample-auth
+```
+{{% /notice %}}
+
+### Deploy auth service
+
+To add this service to your cluster, download the [auth-service yaml](auth-service.yaml) and apply it:
+```
+kubectl apply -f auth-service.yaml
+```
+This file contains the deployment, service and upstream definitions.
+
+## Deploy Gloo and the petstore demo app
+
+Install Gloo-enterprise (version v0.13.5 or above) and the petstore demo:
+```shell
+glooctl install gateway --license-key <YOUR KEY>
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/master/example/petstore/petstore.yaml
+```
+
+Add a route and test that everything so far works:
+
+```shell
+glooctl add route --name default --namespace gloo-system --path-prefix / --dest-name default-petstore-8080 --dest-namespace gloo-system
+URL=$(glooctl proxy url)
+curl $URL/api/pets/
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+## Configure Gloo to use your server
+
+### Configure Gloo settings
+
+Edit the gloo settings (`kubectl edit settings -n gloo-system default`) to point to your auth server. settings should look like this:
+
+{{< highlight yaml "hl_lines=9-18" >}}
+apiVersion: gloo.solo.io/v1
+kind: Settings
+metadata:
+  name: default
+  namespace: gloo-system
+spec:
+  bindAddr: 0.0.0.0:9977
+  discoveryNamespace: gloo-system
+  extensions:
+    configs:
+      extauth:
+        extauthzServerRef:
+          name: auth-server
+          namespace: gloo-system
+        httpService: {}
+        requestBody:
+          maxRequestBytes: 10240
+        requestTimeout: 0.5s
+      rate-limit:
+        ratelimit_server_ref:
+          name: rate-limit
+          namespace: gloo-system
+  kubernetesArtifactSource: {}
+  kubernetesConfigSource: {}
+  kubernetesSecretSource: {}
+  refreshRate: 60s
+{{< /highlight >}}
+
+{{% notice tip %}}
+When using a gRPC auth service, remove the `httpService: {}` line from the config above.
+{{% /notice %}}
+
+This configuration also sets other configuraiton parameters:
+
+- requestBody - When set to, the request body will also be sent to the auth service. with this configuration, a body up to 10KB will be buffered and sent to the auth-service. This is useful in use cases where the auth service needs to compute an HMAC on the body.
+- requestTimeout - A timeout for the auth service response. If the service takes longer to response, the request will be denied.
+
+### Configure the virtual service
```

Comment:
Can we say "VirtualService" here since we're referring to Gloo's `VirtualService` resource?